packages/ruleset/src/utils/get-schema-name-at-path.js

```diff
@@ -42,17 +42,22 @@ function getSchemaNameAtPath(path, pathToReferencesMap) {
   // pathToReferencesMap (which comes from graph nodes that Spectral gives us).
   // See the function documentation above for more info.
   let pathBuilder = '';
-  for (const pathSegment of path.split('.')) {
+  const pathSegments = path.split('.');
+
+  for (let i = 0; i < pathSegments.length; i++) {
```

Comment:
The only real change is that now we use a numerically-indexed for-loop so that we know for sure when we're going through it for the last time.